### PR TITLE
chore: cleanup any lingering instances

### PR DESCRIPTION
**Pull request by tonsV2**
@radnov and I have discussed the need for some automatic removal of leftover instances in regards to failing stack tests.

Ideally the tests shouldn't fail but we need more experience to adjust the thresholds so they don't.

The downside of this PR is that we might remove the evidence but on the other hand, if the problem persists, the tests could be ran again locally.

### CODE REVIEW

File: scripts/stack_tests.sh [hunk 1]
```diff
@@ -7,6 +7,18 @@ GROUP=whoami
...
+  # shellcheck disable=SC2046
+  ./destroy.sh whoami $(./list.sh | jq -r '.[].Instances[]?.Name' | grep $INSTANCE_PREFIX | grep "$INSTANCE_POSTFIX")
+  # shellcheck disable=SC2046
+  ./destroy.sh whoami $(./listPresets.sh | jq -r '.[].Instances[]?.Name' | grep $INSTANCE_PREFIX | grep "$INSTANCE_POSTFIX")
+  trap - EXIT
+  exit
+}
+
+trap cleanup_handler INT
+trap cleanup_handler EXIT
```
Comment (radnov) [inline scripts/stack_tests.sh:20]: The two signals can be `trap`ped on a single line.

**Review by radnov — COMMENTED** [review]
Looks good. Added a small suggestion 👍

File: scripts/stack_tests.sh [hunk 2]
```diff
@@ -7,6 +7,18 @@ GROUP=whoami
 INSTANCE_PREFIX=im-e2e
 INSTANCE_POSTFIX=$(tr -dc '[:lower:]' </dev/urandom | head -c 5; echo '')
 
+function cleanup_handler {
+  # shellcheck disable=SC2046
+  ./destroy.sh whoami $(./list.sh | jq -r '.[].Instances[]?.Name' | grep $INSTANCE_PREFIX | grep "$INSTANCE_POSTFIX")
```
Comment (teleivo) [inline scripts/stack_tests.sh:12]: Just to make sure. I have not executed the bash fully in my head 😂 
Is there any chance we accidentally destroy the wrong instances?

Comment (tonsV2) [reply scripts/stack_tests.sh:12]: Not unless you create an instance with the same pre- and postfix

Comment (tonsV2) [reply scripts/stack_tests.sh:12]: To elaborate a bit more here, we have a fairly specific prefix. Even if we ignore the prefix and only focus on the postfix the following should be the case. The postfix consists of 5 lower case alphabetic characters, which gives us 26^5 (== 11881376) combinations. It should be reasonably safe.

**Review by teleivo — APPROVED** [review]

**Review by radnov — APPROVED** [review]

**Pull request merged**